std/math.d

```diff
@@ -6798,3 +6799,205 @@ real yl2xp1(real x, real y) @nogc @safe pure nothrow;       // y * log2(x + 1)
+ *          -0.0, which is less than +0.0;)
+ *      $(LI If the sign bit is set (that is, it's 'negative'), $(NAN) is less
+ *          than any number; if the sign bit is not set (it is 'positive'),
+ *          $(NAN) is greater than any number;)
+ *      $(LI $(NAN)s of the same sign are ordered by the payload ('negative'
+ *          ones - in reverse order).)
+ * )
+ *
+ * Returns:
+ *      -1 if $(D x) precedes $(D y) in the order specified above;
+ *      0 if $(D x) and $(D y) are identical, and 1 otherwise.
+ *
+ * See_Also:
+ *      $(MYREF isIdentical)
+ * Standards: Conforms to IEEE 754-2008
+ */
+int cmp(T)(T x, T y) @nogc @trusted pure nothrow
+    if (isFloatingPoint!T)
+{
+    alias F = floatTraits!T;
+
+    static if (F.realFormat == RealFormat.ieeeSingle
+               || F.realFormat == RealFormat.ieeeDouble)
+    {
+        static if (T.sizeof == 4)
+            alias UInt = uint;
+        else
+            alias UInt = ulong;
+
+        enum msb = ~(UInt.max >>> 1);
+
+        UInt*[2] vars = [cast(UInt*)&x, cast(UInt*)&y];
+
+        foreach (var; vars)
+            if (*var & msb)
+                *var = ~*var;
+            else
+                *var |= msb;
+
+        if (*vars[0] < *vars[1])
+            return -1;
+        else if (*vars[0] > *vars[1])
+            return 1;
+        else
+            return 0;
+    }
+    else static if (F.realFormat == RealFormat.ieeeExtended53
+                    || F.realFormat == RealFormat.ieeeExtended
+                    || F.realFormat == RealFormat.ieeeQuadruple)
+    {
+        static if (F.realFormat == RealFormat.ieeeQuadruple)
+            alias RemT = ulong;
+        else
+            alias RemT = ushort;
+
+        enum shift = ulong.sizeof / RemT.sizeof;
+
+        ubyte*[2] bytes = [cast(ubyte*)&x, cast(ubyte*)&y];
+
+        ulong*[2] bulk = [cast(ulong*)&x, cast(ulong*)&y];
+        RemT*[2] rem = [cast(ushort*)&x + shift, cast(ushort*)&y + shift];
+
+        foreach (i; 0 .. 2)
+            if (bytes[i][F.SIGNPOS_BYTE] & 0x80)
+            {
+                *bulk[i] = ~*bulk[i];
+                *rem[i] = ~*rem[i];
+            }
+            else
+            {
+                bytes[i][F.SIGNPOS_BYTE] |= 0x80;
+            }
+
+        version(LittleEndian)
+        {
+            alias major = rem;
+            alias minor = bulk;
+        }
+        else
+        {
+            alias major = bulk;
+            alias minor = rem;
+        }
+
+        if (*major[0] < *major[1])
+            return -1;
+        else if (*major[0] > *major[1])
+            return 1;
+        else if (*minor[0] < *minor[1])
+            return -1;
+        else if (*minor[0] > *minor[1])
+            return 1;
+        else
+            return 0;
+    }
+    else
+    {
+        // IBM Extended doubledouble does not follow the general
+        // sign-exponent-significand layout, so has to be handled generically
```

Comment:
Well, this branch isn't actually specific to IBM extended format, it is more of
'catch-all' generic handler akin to last `else` block in `isNaN`.